lib/decanter/parser/boolean_parser.rb

```diff
@@ -5,9 +5,27 @@ class BooleanParser < ValueParser
       allow TrueClass, FalseClass
 
       parser do |val, options|
-        raise Decanter::ParseError.new 'Expects a single value' if val.is_a? Array
+        single_value_check(val)
         next if (val.nil? || val === '')
-        [1, '1'].include?(val) || !!/true/i.match(val.to_s)
+        [1, '1'].include?(val) || !!/true/i.match(val.to_s) || parse_options(val, options)
+      end
+
+      def self.parse_options(val, options)
+        raw_true_value = options.fetch(:true_value, nil)
+        return false if raw_true_value.nil?
+
+        single_value_check(raw_true_value)
+        true_value = raw_true_value.to_s.downcase
+
+        # want to avoid using values that are already implemented
+        accounted_for_values = ['false', 'true', '1', '0', '']
+        return false if accounted_for_values.include?(true_value)
```

Comment:
Interesting. I think I follow the intent here but the current implementation feels misleading or a potential for unexpected behavior (what if someone meant to have "false" as a truth value - idk why just playing it out)

Perhaps an alternative that you're starting to go down that potentially drys things out is to capture some "default" truthy values and the options get appended. Then you could check if the val is included in the whole array of options.  Not sure if anything gets lost from a regex perspective

Reply:
@inveterateliterate appending the option value to a set of default true values definitely cleaned things up a bit and makes the purpose of the option value more clear. good call!